Snake/BusinessLayer/maze.cs

```diff
@@ -40,20 +52,87 @@ public maze(int w,int h)
                 {
                     for (int j = 0; j < width; j++)
                     {
-                        maze[i, j] = border;
-                    } 
+                        Maze[i, j] = BORDER;
+                    }
                 }
 
                 //first and last colum of other rows should be marked as border
                 else
                 {
-                    maze[i, 0] = border;
-                    maze[i, width - 1] = border;
+                    Maze[i, 0] = BORDER;
+                    Maze[i, width - 1] = BORDER;
                 }
             }
 
-            return maze;
         }
 
+
+
+        public void Generate_All_Obstacles()
+        {
+            int Vertical_Line_Head_X = random_number.Next(0,height);
+            int Vertical_Line_Head_Y = random_number.Next(width/2,width);
+
+            Generate_Vertical_Line_Obstacle(Vertical_Line_Head_X,Vertical_Line_Head_Y,random_number.Next(0,height - Vertical_Line_Head_X));
+
+            int Horizon_Line_Head_X = random_number.Next(0,height);
+            int Horizon_Line_Head_Y = random_number.Next(0,width/2 - 1);
+
+            Generate_Horizon_Line_Obstacle(Horizon_Line_Head_X,Horizon_Line_Head_Y,random_number.Next(0,Horizon_Line_Head_Y));
+
+
+            int Rectangle_Head_X = random_number.Next(0, height/2);
+            int Rectangle_Head_Y = random_number.Next(0, width/2);
+
+            int Rectangle_width = random_number.Next(0,width/2 - Rectangle_Head_Y);
+            int Rectangle_height = random_number.Next(0, height/2 - Rectangle_Head_X);
+
+            Generate_Rectangle_Obstacle(Rectangle_Head_X, Rectangle_Head_Y, Rectangle_width, Rectangle_height);
+
+
+
+        }
+
+
+        public void Generate_Horizon_Line_Obstacle(int PointX, int PointY, int LineLength)
```

Comment:
This method is very similar to the Generate_Vertical_Line_Obstacle method - would it be better to have a single method with a direction parameter? i.e. GenerateLineObstacle(pointX, pointY, length, direction) Direction can be Horizontal or Vertical.

Can then do a switch on the direction to add to X or take away from Y.